### PR TITLE
feat(es/minifier): Improve simplification of `?.`

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

**Related issue:**

 - Closes https://github.com/swc-project/swc/issues/6492.

### CODE REVIEW

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_minifier

File: crates/swc_ecma_minifier/tests/fixture/issues/6492/output.js [hunk 1]
```diff
@@ -0,0 +1 @@
+console.log('val', 42);
```
Comment (kdy1) [inline crates/swc_ecma_minifier/tests/fixture/issues/6492/output.js:1]: [Previous output](https://play.swc.rs/?version=1.3.23&code=H4sIAAAAAAAAA0vOzysuUchPylKwVahWSEvMS660UjAxUqi15koGS5Ul5gClgArs9aLzSnNyFGpqFNTB6tRjwUryc1L1cvLTNdSBKtV1QOo1AfhU17pVAAAA&config=H4sIAAAAAAAAA21TS7IbIQy8C2svUl5k8Q6QXVY5AIVBjPUyoCkk%2FDzlenePmLEZj50darob%2FbiZT%2Fbm42YmVxhKO%2FGcxV3Nh5F5AvYFJzEHI6yQlArfei4uc6SSGr2A87IcahZMoEJXhZIT9KrDNFGRP1SLb1fTSj%2BYABcYaUqQ5e7bjF0ZQGMDfPxxPCptJGLVRTcyHEzCjHFuj3lKasXcziqqzYc7z5VCX7z6HsyJaATN%2BDW2ji1mgQHKJvU0jm5isBdXuqK95goy5R1UBYKdCk1PaA4oSFnNHlgAF6ynAB3AAl7wAhtFPZSSWRPrmSxggFMdhjaYlQkXN1Yn3QuuSxv0xa47E7LYWDO%2FQLtMV2gt8s7CaAtILfnB%2BSTMuz78BdA8R8ecXYJNueBR57Bnxv%2BwMEcdoswd1QlvWWUYtDiLGHuFLWsoglu3CoTqoVXoN%2FM7uCuIMYCFGLXbXcxfKP68mbUlp9hD7ZqL2wzW0D72aY%2B2FXgDf2lW8jyOO67%2F4fyK8ZxONL5ZJJAzhTdYGyH0Chbd3uv0itYcQNsM4emicgv7murwhewIsvVZF0I1dhjp1NdXP2Vyeehr2X5polAbcFuat%2FzWn%2BZ7%2B52rH%2FLvO2%2Fx%2BQd0Pd7PaQQAAA%3D%3D)

File: crates/swc_ecma_minifier/tests/fixture/issues/6492/input.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+const obj = { key: 42 };
+const val = obj?.[null || 'key']
```
Comment (jridgewell) [inline crates/swc_ecma_minifier/tests/fixture/issues/6492/input.js:2]: Can we include a few deeper variations?

```js
obj?.key.toString()
obj?.key?.toString()
obj.key?.toString()
```

**Review by jridgewell — CHANGES_REQUESTED** [review]
Nit: The PR title references optional coalescing `??`, but this is improving optional chaining `?.`.

File: crates/swc_ecma_minifier/tests/fixture/issues/6492/input.js (hunk 2 above)
Comment (kdy1) [reply crates/swc_ecma_minifier/tests/fixture/issues/6492/input.js:2]: I'll do, thanks!

**Review by jridgewell — APPROVED** [review]

**Pull request merged**